src/software/amazon/ion/impl/IonReaderTextSystemX.java

```diff
@@ -72,6 +73,46 @@ public IonSystem getSystem()
         return _system;
     }
 
+    // TODO getIntegerType() is duplicated in IonReaderBinarySystemX. It could
+    // be consolidated into a single location, but that would have to be part
```

Comment:
After changing to a callback, this is now stateful (because a method has to be called on this reader instance), so it's less easy to move it into PrivateScalarConversions. It feels like a misuse of that class to put a static method in there that takes an IonReader as an argument, so I've refrained from doing that and kept this comment here.